### PR TITLE
Track secrets in consumer reconciler 

### LINKED ISSUE #2144: Kafka source should pick up missing secret
**Describe the bug**
When a Kafka source is created referencing a missing secret resource, when someone creates the secret the Kafka source references, Kafka source should be able to pick up the secret and becomes ready. 

 **Expected behavior**
Kafka source should be able to pick up the missing secret and becomes ready. 

 **To Reproduce**
1. Create a Kafka source referencing a secret that does not exist. 
2. Create the missing secret and observe that Kafka source does not pick up the secret and never gets ready. 

 **Knative release version**

 **Additional context**
Add any other context about the problem here such as proposed priority

### PR DESCRIPTION
**Pull request by pierDipi**
Fixes #2144 

## Proposed Changes

- Track secrets in consumer reconciler

### CODE REVIEW

**Review by aavarghese — APPROVED** [review]
/lgtm

**Comment by pierDipi** [comment]
@aavarghese there were conflict, I rebased now

File: control-plane/pkg/reconciler/consumer/consumer.go [hunk 1]
```diff
@@ -204,15 +209,6 @@ func (r Reconciler) reconcileAuth(ctx context.Context, c *kafkainternals.Consume
 				Version:   secret.ResourceVersion,
 			},
 		}
-		ref := tracker.Reference{
-			APIVersion: c.APIVersion,
-			Kind:       c.Kind,
-			Namespace:  c.GetNamespace(),
-			Name:       c.GetName(),
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer.go:211]: hmm why was this Reference tracker even here before?

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Review by matzew — APPROVED** [review]
/lgtm
/approve

File: control-plane/pkg/reconciler/consumer/consumer.go (hunk 1 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/consumer/consumer.go:211]: previously, we were only tracking the case where there is only one secret reference (all cases except the KafkaSource one), we're now tracking properly both cases.

**Pull request merged**